File: simpadu-keuangan/app/Models/KeuanganMahasiswa.php
<?php
namespace App\Models;
use Illuminate\Database\Eloquent\Model;

class KeuanganMahasiswa extends Model
{
    protected $table = 'KEUANGAN_MAHASISWA';
    protected $primaryKey = 'ID_KEUANGAN_MHS';
    public $incrementing = false;
    protected $keyType = 'string';
    public $timestamps = false;

    protected $fillable = [
        'ID_KEUANGAN_MHS',
        'ID_KATEGORI',
        'ID_BEASISWA',
        'ID_MAHASISWA',
        'SEMESTER',
        'STATUS_AKTIF',
    ];

    public function kategoriUkt()
    {
        return $this->belongsTo(KategoriUkt::class, 'ID_KATEGORI', 'ID_KATEGORI');
    }

    public function beasiswa()
    {
        return $this->belongsTo(Beasiswa::class, 'ID_BEASISWA', 'ID_BEASISWA');
    }

    public function tagihan()
    {
        return $this->hasMany(Tagihan::class, 'ID_KEUANGAN_MHS', 'ID_KEUANGAN_MHS');
    }
}

File: mobile/lib/models/keuangan_mahasiswa.dart
class KeuanganMahasiswa {
  String idKeuanganMhs;
  String idKategori;
  String idMahasiswa;
  String semester;
  String beasiswa;
  String statusAktif;

  KeuanganMahasiswa({
    required this.idKeuanganMhs,
    required this.idKategori,
    required this.idMahasiswa,
    required this.semester,
    required this.beasiswa,
    required this.statusAktif,
  });

  factory KeuanganMahasiswa.fromJson(Map<String, dynamic> json) {
    return KeuanganMahasiswa(
      idKeuanganMhs: json['id_keuangan_mhs']?.toString() ?? '',
      idKategori: json['id_kategori']?.toString() ?? '',
      idMahasiswa: json['id_mahasiswa']?.toString() ?? '',
      semester: json['semester']?.toString() ?? '',
      beasiswa: json['beasiswa']?.toString() ?? '',
      statusAktif: json['status_aktif']?.toString() ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id_keuangan_mhs': idKeuanganMhs,
      'id_kategori': idKategori,
      'id_mahasiswa': idMahasiswa,
      'semester': semester,
      'beasiswa': beasiswa,
      'status_aktif': statusAktif,
    };
  }
}
